src/libguac/guacamole/argv-constants.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef GUAC_ARGV_CONSTANTS_H
+#define GUAC_ARGV_CONSTANTS_H
+
+/**
+ * Constants related to automatic handling of received "argv" instructions.
+ *
+ * @file argv-constants.h
+ */
+
+/**
+ * Option flag which declares to guac_argv_register() that the associated
+ * argument should be processed exactly once. If multiple "argv" streams are
+ * received for the argument, only the first such stream is processed.
+ * Additional streams will be rejected.
+ */
+#define GUAC_ARGV_OPTION_ONCE 1
+
+/**
+ * Option flag which declares to guac_argv_register() that the values received
+ * and accepted for the associated argument should be echoed to all connected
+ * users via outbound "argv" streams.
+ */
+#define GUAC_ARGV_OPTION_ECHO 2
```

Comment:
With this option, here and implemented in the new `libguac/argv.c` file, is there still a need for the following functions:
`guac_telnet_send_current_argv()`
`guac_ssh_send_current_argv()`
`guac_kubernetes_send_current_argv()`

?  It seems like the echo here and the `guac_*_send_current_argv()` have the same purpose - sending the argv value back to the client.  Is there a reason they are both required?

Reply:
Yep - the current value of the parameters that users may change needs to be sent to the user when the connection starts, as well as to new users joining a shared connection. This is distinct from the case where an update to a parameter is broadcast to all current users of the connection (`GUAC_ARGV_OPTION_ECHO`).